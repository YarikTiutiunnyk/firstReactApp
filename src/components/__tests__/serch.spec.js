import React from 'react';
import { shallow, mount } from 'enzyme';
import SerchBar from '../serch';

describe('SerchBar test', () => {
  const mocksetSerchString = jest.fn();
  const mocksetItemsError = jest.fn();
  const mocksetItems = jest.fn();
  const props = {
    itemFound: '0',
    textError: '',
    type: 'movie',
    setSerchString: mocksetSerchString,
    setItemsError: mocksetItemsError,
    setItems: mocksetItems
  };
  //const initialState = { input: 'hello' };

  describe('SerchBar initial', () => {
    const wrapper = shallow(<SerchBar {...props} />);

    it('renders properly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('change state', () => {
      wrapper.setState({ input: 'hello' });
      //console.log(wrapper.state());
      expect(wrapper.state('input')).toEqual('hello');
    });
  });

  describe('Samulate events', () => {
    const wrapper = shallow(<SerchBar {...props} />);
    const newText = 'new text';

    it('change on input form', () => {
      wrapper.find('Input').simulate('change', {
        target: {
          value: newText
        }
      });
      expect(wrapper.state('input')).toEqual(newText);
    });

    it('click on the button', () => {
      const wrapper = mount(<SerchBar {...props} />);
      //console.log(wrapper.find('Button').debug());
      wrapper.find('Button').simulate('click');
      expect(mocksetSerchString).toHaveBeenCalledTimes(1);
      wrapper.unmount();
    });
  });
});
