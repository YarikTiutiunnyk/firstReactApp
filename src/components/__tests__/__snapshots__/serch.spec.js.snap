// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SerchBar test SerchBar initial renders properly 1`] = `
<div
  className="SerchContainer"
>
  <Input
    action={
      <Button
        as="button"
        className="lll"
        icon="search"
        onClick={[Function]}
      />
    }
    disabled={false}
    fluid={true}
    onChange={[Function]}
    placeholder="Search..."
    size="large"
    type="text"
    value=""
  />
  <Header
    as="h4"
    content="We found 0 item"
    subheader=""
  />
</div>
`;
